[BUG] CMake does not work with Visual Studio

Please prefix your issue with one of the following: [BUG], [PROPOSAL], [QUESTION].

CCExtractor version (using the --version parameter preferably) : 0.85

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/ccextractor/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/ccextractor/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.
- [X] I have used the latest available version of CCExtractor to verify this issue exists.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never used CCExtractor.
- [X] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [ ] I am an active contributor to CCExtractor.

**Necessary information**
- Is this a regression (did it work before)? [ ] NO | [ ] YES - *please specify the last known working version*
- What platform did you use? [X] Windows - [ ] Linux - [ ] Mac
- What where the used arguments? N/A

**Video links**

N/A

**Additional information**

CMake does not work with Visual Studio. Following errors raises when I was trying to compile it using the .sln file generated by CMake:
```
1>base_encoding.c
1>d:\gitprojects\ccextractor\src\zlib\zconf.h(452): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
1>bitstream.c
1>mp4.c
1>d:\gitprojects\ccextractor\src\lib_ccx\stdintmsc.h(90): error C2371: 'int_fast16_t': redefinition; different basic types
1>d:\gitprojects\ccextractor\src\lib_ccx\stdintmsc.h(94): error C2371: 'uint_fast16_t': redefinition; different basic types
1>os_divers.c
1>D:\GitProjects\ccextractor\src\gpacmp4\os_divers.c(57): fatal error C1083: Cannot open include file: 'sys/time.h': No such file or directory
1>os_file.c
1>D:\GitProjects\ccextractor\src\gpacmp4\os_file.c(44): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
2>cl : Command line error D8021: invalid numeric argument '/Wno-write-strings'
2>Done building project "ccextractor.vcxproj" -- FAILED.
```
CMake version used: 3.7.2

However, windows\ccextractor.sln works.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -75,7 +75,7 @@ endif(WIN32)
 if(MSVC)
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -W3 /wd4005 /wd4996")
 else (MSVC)
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -O0 -Wno-write-strings -g -std=gnu99")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wno-write-strings -g -std=gnu99")
 endif(MSVC)
 
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_FILE_OFFSET_BITS=64")

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -62,21 +62,28 @@ aux_source_directory ("${PROJECT_SOURCE_DIR}/zlib/" SOURCEFILE)
 
 #Windows specific libraries and linker flags
 if(WIN32)
+    include_directories ("${PROJECT_SOURCE_DIR}/win_spec_incld/")
     include_directories ("${PROJECT_SOURCE_DIR}/win_iconv/")
     aux_source_directory ("${PROJECT_SOURCE_DIR}/win_iconv/" SOURCEFILE)
-    set (EXTRA_LIBS ${EXTRA_LIBS} -lws2_32 -lwinmm)
+    set (EXTRA_LIBS ${EXTRA_LIBS} ws2_32 winmm)
 else (WIN32)
     # Adding some platform specific library path
     link_directories (/opt/local/lib)
     link_directories (/usr/local/lib)
 endif(WIN32)
 
-set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0 -Wall -g -std=gnu99 -Wno-write-strings -D_FILE_OFFSET_BITS=64")
+if(MSVC)
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -W3 /wd4005 /wd4996")
+else (MSVC)
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -O0 -Wno-write-strings -g -std=gnu99")
+endif(MSVC)
+
+set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_FILE_OFFSET_BITS=64")
 add_subdirectory (lib_ccx)
 
 aux_source_directory (${PROJECT_SOURCE_DIR} SOURCEFILE)
 set (EXTRA_LIBS ${EXTRA_LIBS} ccx)
-set (EXTRA_LIBS ${EXTRA_LIBS} m)
+# set (EXTRA_LIBS ${EXTRA_LIBS} m)
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set (EXTRA_LIBS ${EXTRA_LIBS} iconv)

--- a/src/lib_ccx/CMakeLists.txt
+++ b/src/lib_ccx/CMakeLists.txt
@@ -1,6 +1,14 @@
 cmake_policy (SET CMP0037 NEW)
 
-set (CMAKE_C_FLAGS "-O0 -Wall -g -std=gnu99")
+if(MSVC)
+    set (CMAKE_C_FLAGS "-W3 /wd4005 /wd4996")
+else (MSVC)
+    set (CMAKE_C_FLAGS "-Wall -g -std=gnu99")
+endif(MSVC)
+
+if(WIN32)
+    add_definitions(-DWIN32)
+endif(WIN32)
 
 if (WITH_FFMPEG)
     find_package(PkgConfig)

